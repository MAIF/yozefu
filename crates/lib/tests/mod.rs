use insta::{assert_debug_snapshot, glob};
use std::fs;
use yozefu_lib::{parse_search_query, ExportedKafkaRecord, KafkaRecord};

#[test]
fn test_inputs() {
    glob!("inputs/*.sql", |path| {
        unsafe {
            use std::env;
            // Set the timezone to Paris to have a fixed timezone for the tests
            env::set_var("TZ", "Europe/Paris");
        }

        let input = fs::read_to_string(path).unwrap();
        let input = input.trim();
        insta::with_settings!({
            description => input.replace("\n", " "),
            filters => vec![
            ("[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}\\.[0-9]{6,}\\+[0-9]{2}:[0-9]{2}", "[datetime]"),
        ]}, {
            assert_debug_snapshot!(parse_search_query(input));
        });
    });
}

#[test]
fn test_exported_record() {
    glob!("inputs/record*.json", |path| {
        let input = fs::read_to_string(path).unwrap();
        let record: KafkaRecord = serde_json::from_str(&input).unwrap();
        assert_debug_snapshot!(ExportedKafkaRecord::from(&record));
    });
}
