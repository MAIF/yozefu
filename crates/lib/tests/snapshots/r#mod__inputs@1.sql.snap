---
source: crates/lib/tests/mod.rs
expression: parse_search_query(input)
input_file: crates/lib/tests/inputs/1.sql
---
Ok(
    (
        "",
        SearchQuery {
            expression: OrExpression(
                [
                    AndTerm(
                        Atom(
                            Filter(
                                Filter {
                                    name: "my-wasm-filter",
                                    parameters: [
                                        String(
                                            "cool",
                                        ),
                                        String(
                                            "cat",
                                        ),
                                    ],
                                },
                            ),
                        ),
                    ),
                    AndTerm(
                        Atom(
                            Parenthesis(
                                OrTerm(
                                    AndExpression(
                                        [
                                            Atom(
                                                Compare(
                                                    TimestampBetween(
                                                        2024-05-28T17:55:08.145+02:00,
                                                        [datetime],
                                                    ),
                                                ),
                                            ),
                                            Atom(
                                                Compare(
                                                    Value(
                                                        None,
                                                        Contain,
                                                        "foundation",
                                                    ),
                                                ),
                                            ),
                                        ],
                                    ),
                                ),
                            ),
                        ),
                    ),
                ],
            ),
            limit: Some(
                1000,
            ),
            from: Some(
                Beginning,
            ),
            order_by: OrderBy {
                order: Key,
                keyword: Desc,
            },
        },
    ),
)
